crates/e2e/tests/e2e/eth_flow.rs

```diff
@@ -0,0 +1,119 @@
+use crate::{deploy::Contracts, tx_value};
+use contracts::{CoWSwapEthFlow, WETH9};
+use ethcontract::{transaction::TransactionResult, Account, Bytes, H160, H256};
+use hex_literal::hex;
+use model::{
+    order::{Order, OrderBuilder, OrderKind},
+    quote::OrderQuoteResponse,
+    signature::hashed_eip712_message,
+    DomainSeparator,
+};
+use refunder::{
+    ethflow_order::EthflowOrder,
+    refund_service::{INVALIDATED_OWNER, NO_OWNER},
+};
+
+pub struct ExtendedEthFlowOrder(pub EthflowOrder);
+
+impl ExtendedEthFlowOrder {
+    pub fn from_quote(quote_response: &OrderQuoteResponse, valid_to: u32) -> Self {
+        let quote = &quote_response.quote;
+        ExtendedEthFlowOrder(EthflowOrder {
+            buy_token: quote.buy_token,
+            receiver: quote.receiver.expect("eth-flow order without receiver"),
+            sell_amount: quote.sell_amount,
+            buy_amount: quote.buy_amount,
+            app_data: ethcontract::Bytes(quote.app_data.0),
+            fee_amount: quote.fee_amount,
+            valid_to, // note: valid to in the quote is always unlimited
+            partially_fillable: quote.partially_fillable,
+            quote_id: quote_response.id.expect("No quote id"),
+        })
+    }
+
+    fn to_cow_swap_order(&self, ethflow: &CoWSwapEthFlow, weth: &WETH9) -> Order {
+        // Each ethflow user order has an order that is representing
+        // it as EIP1271 order with a different owner and valid_to
+        OrderBuilder::default()
+            .with_kind(OrderKind::Sell)
+            .with_sell_token(weth.address())
+            .with_sell_amount(self.0.sell_amount)
+            .with_fee_amount(self.0.fee_amount)
+            .with_receiver(Some(self.0.receiver))
+            .with_buy_token(self.0.buy_token)
+            .with_buy_amount(self.0.buy_amount)
+            .with_valid_to(u32::MAX)
+            .with_app_data(self.0.app_data.0)
+            .with_eip1271(ethflow.address(), hex!("").into())
+            .build()
+    }
+
+    pub fn include_slippage_bps(&self, slippage: u16) -> Self {
+        let max_base_point = 10000;
```

Comment:
```suggestion
        const MAX_BASE_POINTS: usize = 10_000usize;
```

As a side note we have a few different `Slippage` types which implement conversions from percent to bps. Eventually we should only have a single one and use them instead of applying the slippage by hand every time.